src/test/java/graphql/annotations/GraphQLExtensionsTest.java

```diff
@@ -0,0 +1,86 @@
+/**
+ * Copyright 2016 Yurii Rashkovskii
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ */
+package graphql.annotations;
+
+import graphql.ExecutionResult;
+import graphql.GraphQL;
+import graphql.schema.GraphQLFieldDefinition;
+import graphql.schema.GraphQLObjectType;
+import graphql.schema.GraphQLSchema;
+import org.testng.annotations.AfterTest;
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.Test;
+
+import java.util.List;
+import java.util.Map;
+
+import static graphql.Scalars.GraphQLString;
+import static graphql.schema.GraphQLSchema.newSchema;
+import static org.testng.Assert.assertEquals;
+
+public class GraphQLExtensionsTest {
+
+    @GraphQLDescription("TestObject object")
+    @GraphQLName("TestObject")
+    private static class TestObject {
+        @GraphQLField
+        public
+        String field() {
+            return "test";
+        }
+
+    }
+
+    @GraphQLTypeExtension(GraphQLExtensionsTest.TestObject.class)
+    private static class TestObjectExtension {
+        @GraphQLField
+        public static String field2() {
+            return "test2";
+        }
+    }
+
+    @Test
+    public void fields() {
+        GraphQLAnnotations.getInstance().registerTypeExtension(TestObjectExtension.class);
+        GraphQLObjectType object = GraphQLAnnotations.object(GraphQLExtensionsTest.TestObject.class);
+        GraphQLAnnotations.getInstance().unregisterTypeExtension(TestObjectExtension.class);
```

Comment:
Hi, 

For 1/, without my changes, so without the annotation, and by calling `GraphQLAnnotations.object(GraphQLExtensionsTest.TestObjectExtension.class)`, you will basically create a new type, named TestObjectExtension, with all fields from TestObject and TestObjectExtension. The TestObject type created by `GraphQLAnnotations.object(GraphQLExtensionsTest.TestObject.class)` won't be impacted.

2/ I was looking for some solution like what you propose, but I thought it was not quite clear as the type returned in that case is not intended to be a usable type, more a list of things to add to an existing type. Actually, I should maybe not create a type from this class at all by calling `getObjectBuilder()`, but only call `getField()` on all fields and methods when the extension is needed.
If we want to use object(), you'll have to return a type, and you'll expect to be usable type. I like the idea of having one single method to register any annotated class and let GraphQLAnnotations do whatever it needs to do with it, but here I don't have any type to return. That's why I added a `registerTypeExtension()` method, which does not return anything.

What do you think ?